src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/StorageWatcher.cs

```diff
@@ -93,6 +97,21 @@ public virtual async Task CreateOrUpdatePackage(Package pkg)
             try
             {
                 await gitHubClient.CreateOrUpdateFiles(pkg.GetCommitContents(), pkg.GetCommitMessage());
+                // Insert Kusto Cluster Mapping to Configuration table.
+                if(pkg.GetCommitContents().Any(content => content.FilePath.Contains("kustoClusterMappings", StringComparison.CurrentCultureIgnoreCase)))
+                {
+                    var kustoMappingPackage = pkg.GetCommitContents().Where(c => c.FilePath.Contains("kustoClusterMappings", StringComparison.CurrentCultureIgnoreCase)).FirstOrDefault();
+                    var githubCommit = await gitHubClient.GetCommitByPath(kustoMappingPackage.FilePath);
+                    // store kusto cluster data
+                    var diagconfiguration = new DiagConfiguration{
+                        PartitionKey = "KustoClusterMapping",
+                        RowKey = kustoMappingPackage.FilePath.Split("/kustoClusterMappings.json")[0],
```

Comment:
You can also get the same value by using `pkg.Id.ToLower()` instead of `kustoMappingPackage.FilePath.Split("/kustoClusterMappings.json")[0]`

Reply:
Does `pkg.Id.ToLower()` give something like `microsoftwebconfiguration` ?

Reply:
Yes. Judging by looking at https://github.com/Azure/Azure-AppServices-Diagnostics/blob/main/src/Diagnostics.RuntimeHost/Controllers/Configuration/KustoClusterMappingsController.cs#L38

Reply:
I see it now! Thanks Hawk